### ISSUE TITLE
BUG in routes

### ISSUE THREAD

**Issue by theperfectcoder** [tiangolo/fastapi#5103]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter

from app.api.api_v1.endpoints.api import users, hotels, gifts
from app.api.api_v1.endpoints.api.auth import auth

api_router = APIRouter(
    tags=['API']
)

api_router.include_router(auth.router)
api_router.include_router(users.router)
api_router.include_router(hotels.router)
api_router.include_router(gifts.router)
```

### Description

I have a bug in Fast API
in my hotels routes I have a route which is not require for authentication but it also returning error "Not authenticated",

I debuged that bug and I found that in another routes have a routes requires auth and when I commented another routes (auth, users, gifts) my route started to work correctly

### Operating System

macOS

### Operating System Details

M1 pro

### FastAPI Version

0.54.2

### Python Version

Python 3.9.7

### Additional Context

.

**Comment by odiseo0** [comment]
I don't think this is a bug because of FastAPI, but since you're using a version that may be too old I think you should search if anyone have ever faced the same bug

**Comment by JarroVGIT** [comment]
It would be most helpful if you provide more information, as this is not really a bug report. "I see a bug, if I do X then Y happens but when I do Z then it doesn't happen" is really not enough to go on or to help you. 

For example, how is authentication handled? Is it a dependency on one of the sub routers, and is it persisted on sibling routers as well? How, why, what did you debug and found that makes you state it is a bug in FastAPI? Where in the code goes this wrong?

**Comment by theperfectcoder** [comment]
Thanks for replying

I used  https://github.com/tiangolo/full-stack-fastapi-postgresql for generate FastAPI project and all auth features is default

**Comment by theperfectcoder** [comment]
@router.get('/my-objects')
def index(
        db: Session = Depends(deps.get_db),
) -> Any:
    return 1
here is the code

**Comment by JarroVGIT** [comment]
Please format your code when posting snippets, that makes it readable and copyable. I also don't understand what the context of that one little piece of endpoint is. 

 I would suggest you open an issue in the repository you are trying out the code from. I am still not convinced you found a bug but rather have some configuration mismatch.

**Comment by odiseo0** [comment]
What does the `auth` have? I also think you should start from scratch, doing something simple instead of cloning a repository that is two years old

**Comment by theperfectcoder** [comment]
> @router.get('/my-objects')
> def index(
> db: Session = Depends(deps.get_db),
> ) -> Any:
> return 1
> here is the code



> Please format your code when posting snippets, that makes it readable and copyable. I also don't understand what the context of that one little piece of endpoint is.
> 
> I would suggest you open an issue in the repository you are trying out the code from. I am still not convinced you found a bug but rather have some configuration mismatch.

```
@router.get('/my-objects')
def index(
        db: Session = Depends(deps.get_db),
) -> Any:
    return 1
```

im sorry for that

**Comment by iudeen** [comment]
Adding authentication is simple in FastAPI. I'd suggest you to look into the [documentation](https://fastapi.tiangolo.com/tutorial/security/get-current-user/). You can use the example directly!

**Comment by iammujtaba** [comment]
For authentication simply use JWT token based authentication as mentioned in the docs. And try to avoid using third-party packages. like full stack fastapi etc... Try to design the structure of your own and code it.
@theperfectcoder
